util/http/retry.go

```diff
@@ -18,17 +18,23 @@ package http
 
 import (
 	"context"
+	"fmt"
 	"math/rand"
 	"net"
 	"net/http"
 	"time"
 
 	"github.com/awslabs/soci-snapshotter/config"
+	"github.com/awslabs/soci-snapshotter/version"
 	"github.com/containerd/containerd/log"
 	rhttp "github.com/hashicorp/go-retryablehttp"
 	"github.com/sirupsen/logrus"
 )
 
+var (
+	UserAgent = fmt.Sprintf("soci-snapshotter/%s", version.Version)
```

Comment:
Both the version/revision are set at [link time](https://github.com/awslabs/soci-snapshotter/blob/main/Makefile#L30). Unless, we also add the ld flags to `go test`, the `version.Version` will not be set and will remain " < undefined > " in our unit tests.

Reply:
Good catch. I think it should be the same as it would be for a real build from the same source/commit.

Reply:
I agree with Clarence. I wonder if we should just have another test that verifies `version.Version != "<undefined>"`

Reply:
Cool then I think we can switch to having ldflags present in go test.